dataland-frontend/src/components/generics/ViewFrameworkBase.vue

```diff
@@ -124,49 +232,95 @@ export default defineComponent({
+      this.dataTypesInDropdown = [];
+      const setOfAvailableFrameworksForCompany = [
+        ...new Set(listOfDataMetaInfo.map((dataMetaInfo) => dataMetaInfo.dataType)),
+      ];
+      const listOfDistinctAvailableAndViewableFrameworksForCompany: string[] = [];
+      setOfAvailableFrameworksForCompany.forEach((dataType) => {
+        if (ARRAY_OF_FRAMEWORKS_WITH_VIEW_PAGE.includes(dataType)) {
+          listOfDistinctAvailableAndViewableFrameworksForCompany.push(dataType);
+        }
+      });
+      listOfDistinctAvailableAndViewableFrameworksForCompany.sort().forEach((dataType) => {
         this.dataTypesInDropdown.push({ label: humanizeString(dataType), value: dataType });
-      }
+      });
     },
 
     /**
-     * Goes through all data meta info for the currently viewed company and does two things. First it saves all distinct
-     * data types into the vue components dataTypesInDropdown-array. Second it collects all data IDs for data of the
-     * currently selected framework type and emits them.
+     * Uses a list of data meta info and filters out all elements whose data type (framework) do not equal the
+     * dataType-prop set for this Vue-component during render.
+     *
+     * @param listOfDataMetaInfo a list of data meta info
+     * @returns the filtered list of data meta info
      */
-    async getAllDataIdsForFrameworkAndEmitThem() {
+    filterListOfDataMetaInfoForChosenFrameworkAndReturnIt(
+      listOfDataMetaInfo: DataMetaInformation[]
+    ): DataMetaInformation[] {
+      return listOfDataMetaInfo.filter((dataMetaInfo) => dataMetaInfo.dataType === this.dataType);
+    },
+
+    /**
+     * Uses a list of data meta info to set a map which has the distinct repoting periods as keys, and the respective
+     * active data meta info as value.
+     * It only takes into account data meta info whose dataType equals the current dataType prop value.
+     *
+     * @param listOfActiveDataMetaInfo The list to be used as input for the map.
+     */
+    setMapOfReportingPeriodToActiveDatasetFromListOfActiveMetaDataInfo(
+      listOfActiveDataMetaInfo: DataMetaInformation[]
+    ) {
+      this.mapOfReportingPeriodToActiveDataset = new Map<string, DataMetaInformation>();
+      listOfActiveDataMetaInfo.forEach((dataMetaInfo: DataMetaInformation) => {
+        if (dataMetaInfo.dataType === this.dataType) {
+          if (dataMetaInfo.currentlyActive) {
+            this.mapOfReportingPeriodToActiveDataset.set(dataMetaInfo.reportingPeriod, dataMetaInfo);
+          } else {
+            throw TypeError("Received inactive dataset meta info from Dataland Backend");
+          }
+        }
+      });
+    },
+
+    /**
+     * Goes through all data meta info for the currently viewed company and does two things.
+     * First it sets the distinct frameworks as options in the framework-dropdown.
+     * Finally it emits a list with data meta info elements for all active datasets for this company. TODO
```

Comment:
TODO here

Reply:
fixed => jsdoc adjusted and TODO removed